src/app/bar-graph/page.tsx

```diff
@@ -49,7 +50,7 @@ export default function Page() {
 						}
 					/>
 					<GridLines {...gridline} />
-					<Bars />
+					{legend.loading ? <BarLoading /> : <Bars />}
```

Comment:
<Bars/> should take a `loading` prop similar to other components that take loading states.
<Bars/> render method should be modified to this


```
// pseudo code that doesn't show rest of props.
export const Bars = ({loading}) => {
   return horizontal ? <BarsHorizontal/> : <BarsVertical loading={loading}/>;
}
// pseudo code that doesn't show rest of props.
export const BarsVertical = ({loading}) => {
     if(loading) return <BarsVerticalLoading/>
}
```